### ISSUE TITLE
GPU: Consider timeout parameter on AcquireSwapchainTexture/Wait/WaitForFences

### ISSUE THREAD

**Issue by thatcosmonaut** [libsdl-org/SDL#10702]
Raising this as a potential ABI issue.

I honestly can't tell if exposing this is worse than not exposing it, but it's conceivable that the client might want to be able to handle these situations instead of stalling forever.

**Comment by slouken** [comment]
For the thread synchronization objects we have SDL_Wait...() and SDL_Wait...WithTimeout()

**Comment by thatcosmonaut** [comment]
Yeah it does sound nice to split them up. If that's the case we can add it even after the freeze.

**Comment by slouken** [comment]
> Yeah it does sound nice to split them up. If that's the case we can add it even after the freeze.

Yep.
